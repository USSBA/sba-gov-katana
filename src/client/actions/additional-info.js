export function reviewAnswers(additionalInfoData) {
  return {
    type: "REVIEW_ANSWERS",
    additionalInfoData: additionalInfoData
  };
}