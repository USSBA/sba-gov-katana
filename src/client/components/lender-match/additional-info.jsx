import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { FormPanel } from '../common/form-styling.jsx'
import { FormGroup, Checkbox, Col } from 'react-bootstrap'
import CheckBox from '../atoms/checkbox.jsx';
import * as LenderMatchActions from '../../actions/lender-match.js';
import * as LocationChangeActions from '../../actions/location-change.js';
import styles from './lender-match.scss';
import addInfoStyles from './additional-info.scss'



export class AdditionalInfoForm extends React.Component {
  constructor(props) {
    super();
    let additionalInfoFields = Object.assign({}, {
      hasWrittenPlan: false,
      hasFinancialProjections: false,
      isGeneratingRevenue: false,
      isVeteran: false
    }, props.additionalInfoFields);
    this.state = {
      additionalInfoFields: additionalInfoFields
    };
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.actions.reviewAnswers(this.state.additionalInfoFields);
    this.props.locationActions.locationChange("/linc/form/review");
    this.addInfoForm.reset()
  }
  ;

  handleClick(e) {
    let newState = {};
    newState[e.target.name] = e.target.checked;
    this.setState({
      additionalInfoFields: {
        ...this.state.additionalInfoFields,
        ...newState
      }
    });
  }

  render() {
    return (
      <div>
        <form ref={ (input) => this.addInfoForm = input } onSubmit={ (e) => this.handleSubmit(e) }>
          <FormGroup>
            <label className={addInfoStyles.label}>Select all that apply to you: </label>
            <CheckBox autoFocus={true} name="hasWrittenPlan" label="I have written a business plan" handleChange={this.handleClick.bind(this)} checked={this.state.additionalInfoFields.hasWrittenPlan}/>
            <CheckBox name="hasFinancialProjections" label="I have financial projections" handleChange={this.handleClick.bind(this)} checked={ this.state.additionalInfoFields.hasFinancialProjections }/>
            <CheckBox name="isGeneratingRevenue" label="I'm generating revenue" handleChange={this.handleClick.bind(this)} checked={ this.state.additionalInfoFields.isGeneratingRevenue }/>
          </FormGroup>
          <button className={ styles.continueBtn } type="submit"> CONTINUE </button>
        </form>
      </div>
    )
  }
}


function mapReduxStateToProps(reduxState) {
  return {
    additionalInfoFields: reduxState.lenderMatch.additionalInfoData
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(LenderMatchActions, dispatch),
    locationActions: bindActionCreators(LocationChangeActions, dispatch)
  }
}
export default connect(
  mapReduxStateToProps,
  mapDispatchToProps
)(AdditionalInfoForm);
