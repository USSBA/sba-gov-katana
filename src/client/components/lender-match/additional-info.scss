.label {
    display: block;
    font-family: "Source Sans Pro", sans-serif;
    font-size: 18px;
    margin: 5px 0 10px 0;
    color: #333333;
}