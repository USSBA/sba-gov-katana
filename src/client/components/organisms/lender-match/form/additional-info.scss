.label {
    display: block;
    margin: 5px 0 10px 0;
    font-weight: normal;
}
