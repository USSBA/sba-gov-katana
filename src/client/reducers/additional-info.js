const additionalInfoReducer = (state = {}, action) => {
  if (action.type === "REVIEW_ANSWERS") {
    return Object.assign({}, state, {
      additionalInfoData: action.additionalInfoData
    });
  }
  return state;
};


export default additionalInfoReducer;