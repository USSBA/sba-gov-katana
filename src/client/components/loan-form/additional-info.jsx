import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { SelectBox, CheckBox } from '../helpers/form-helpers.jsx'
import { FormPanel } from '../common/form-styling.jsx'
import { browserHistory } from 'react-router';

import { FormGroup, Checkbox, Panel, ButtonToolbar, Row } from 'react-bootstrap'

import * as AdditionalInfoActions from '../../actions/additional-info.js'

import styles from '../common/styles.scss';


export class AdditionalInfoForm extends React.Component {
    constructor(){
        super();
        this.state ={
            additionalInfoFields: {
            }
        }
    }

    handleSubmit(e){
        e.preventDefault();
        this.props.actions.reviewAnswers(this.state.additionalInfoFields);
        // console.log(this.state)
        browserHistory.push("/form/review");
        this.addInfoForm.reset()
    };

    handleChange(e){
        let additionalInfoFields = {};
        additionalInfoFields[e.target.name] = e.target.value;
        this.setState({additionalInfoFields: {...this.state.additionalInfoFields, ...additionalInfoFields}})
        // console.log(this.state)
    }

    handleClick(e){
        let additionalInfoFields = {};
        additionalInfoFields[e.target.name] = e.target.value;
        this.setState({additionalInfoFields: {...this.state.additionalInfoFields, ...additionalInfoFields}});
        // console.log(this.state);
    }

    render(){
        return (
            <FormPanel title="Additional Info">
                <form ref={(input) => this.addInfoForm = input} onSubmit={(e) => this.handleSubmit(e)}>

                    <SelectBox
                        label="What's your industry experience?"
                        name = "industryExp"
                        handleChange={this.handleChange.bind(this)}>
                        <option value="" disabled>- Select use of funds -</option>
                        <option defaultValue=""></option>
                        <option value="none">none</option>
                        <option value="lots">lots</option>
                    </SelectBox>

                    <FormGroup className="col-xs-6 col-xs-offset-3">
                        <label className={styles.controlLabel}>Check all that apply to you: </label>
                        <Checkbox name = "hasWrittenPlan" onClick={this.handleClick.bind(this)}>
                            I have a written business plan
                        </Checkbox>
                        <Checkbox name = "hasFinancialProjections" onClick={this.handleClick.bind(this)}>
                            I have financial projections
                        </Checkbox>
                        <Checkbox name="isGeneratingRevenue" onClick={this.handleClick.bind(this)}>
                            I'm generating revenue
                        </Checkbox>
                        <Checkbox name="isVeteran" onClick={this.handleClick.bind(this)}>
                            I'm a veteran
                        </Checkbox>
                    </FormGroup>

                    <button className="btn btn-default col-xs-2 col-xs-offset-5"
                            type="submit">
                        Continue </button>
                </form>
            </FormPanel>
        )
    }
}


function mapReduxStateToProps(reduxState) {
    console.log(reduxState)
    return {};
}

function mapDispatchToProps(dispatch){
    return {
        actions: bindActionCreators(AdditionalInfoActions, dispatch)
    }
}
export default connect(
    mapReduxStateToProps,
    mapDispatchToProps
)(AdditionalInfoForm);
