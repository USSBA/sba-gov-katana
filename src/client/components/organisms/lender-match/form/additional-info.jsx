import React from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import * as LenderMatchActions from '../../../../actions/lender-match.js';
import * as LocationChangeActions from '../../../../actions/location-change.js';
import {logEvent} from "../../../../services/analytics.js";

import CheckBox from '../../../atoms/checkbox/checkbox.jsx';
import FormPageButtons from '../../../molecules/form-page-buttons/form-page-buttons.jsx';
import styles from './lender-match.scss';
import addInfoStyles from './additional-info.scss'

export class AdditionalInfoForm extends React.Component {
  constructor(props) {
    super();
    let additionalInfoFields = Object.assign({}, {
      hasWrittenPlan: false,
      hasFinancialProjections: false,
      isGeneratingRevenue: false,
      isVeteran: false
    }, props.additionalInfoFields);
    this.state = {
      additionalInfoFields: additionalInfoFields
    };
  }

  componentDidMount() {
    window.scrollTo(0, 0);
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.actions.reviewAnswers(this.state.additionalInfoFields);
    this.props.locationActions.locationChange("/linc/form/review", {
      action: "Continue Button Pushed",
      label: "/linc/form/review"
    });
    this.addInfoForm.reset()
  };

  handleClick(e) {
    let newState = {};
    newState[e.target.name] = e.target.checked;
    this.setState({
      additionalInfoFields: {
        ...this.state.additionalInfoFields,
        ...newState
      }
    }, function() {
      logEvent({"category": "Lender Match Form", "action": "Focus Event", "label": e.target.name});
    });
  }

  render() {
    let id = "lender-match-additional-info-form";
    return (
      <div>
        <form id={id} ref={(input) => this.addInfoForm = input} onSubmit={(e) => this.handleSubmit(e)}>
          <h6 className={addInfoStyles.label}>Select all that apply to you:</h6>
          <CheckBox id={id+"-plan"} autoFocus={true} name="hasWrittenPlan" label="I have written a business plan" handleChange={this.handleClick.bind(this)} checked={this.state.additionalInfoFields.hasWrittenPlan}/>
          <CheckBox id={id+"-projections"} name="hasFinancialProjections" label="I have financial projections" handleChange={this.handleClick.bind(this)} checked={this.state.additionalInfoFields.hasFinancialProjections}/>
          <CheckBox id={id+"-revenue"} name="isGeneratingRevenue" label="I'm generating revenue" handleChange={this.handleClick.bind(this)} checked={this.state.additionalInfoFields.isGeneratingRevenue}/>
          <FormPageButtons parentId={id} backButtonHandler={this.props.locationActions.goBack} continueButtonHandler={this.handleSubmit.bind(this)}/>
        </form>
      </div>
    )
  }
}

function mapReduxStateToProps(reduxState) {
  return {additionalInfoFields: reduxState.lenderMatch.additionalInfoData};
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(LenderMatchActions, dispatch),
    locationActions: bindActionCreators(LocationChangeActions, dispatch)
  }
}
export default connect(mapReduxStateToProps, mapDispatchToProps)(AdditionalInfoForm);
