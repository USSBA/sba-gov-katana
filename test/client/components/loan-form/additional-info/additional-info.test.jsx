import React from 'react';
import { AdditionalInfoForm } from '../../../../../src/client/components/lender-match/additional-info.jsx';
import renderer from 'react-test-renderer';

test('AdditionalInfo page renders input fields', () => {
    const component = renderer.create(
        <AdditionalInfoForm />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})