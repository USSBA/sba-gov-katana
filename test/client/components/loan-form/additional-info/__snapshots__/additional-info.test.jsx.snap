exports[`test AdditionalInfo page renders input fields 1`] = `
<div
  className="col-xs-12 col-lg-6 col-lg-offset-3 formPanel panel panel-default"
  id={undefined}>
  <div
    className="panel-body">
    <div
      className="btn-toolbar"
      role="toolbar">
      <button
        className="btn btn-default btn-sm pull-left"
        onClick={[Function]}
        type="button">
        <span
          className="glyphicon glyphicon-chevron-left" />
         Back
      </button>
    </div>
    <h2
      className="col-md-4 col-md-offset-4 text-center">
      Additional Info
    </h2>
    <form
      onSubmit={[Function]}>
      <div
        className="col-sm-6 col-sm-offset-3 col-xs-12 col-xs-offset-0">
        <div
          className="form-group">
          <label
            className="controlLabel control-label"
            htmlFor={undefined}>
            What\'s your industry experience?
          </label>
          <select
            className="form-control"
            id={undefined}
            name="industryExp"
            onChange={[Function]}
            type={undefined}>
            <option
              defaultValue="" />
            <option
              value="none">
              none
            </option>
            <option
              value="lots">
              lots
            </option>
          </select>
        </div>
      </div>
      <div
        className="col-xs-6 col-xs-offset-3 form-group">
        <label
          className="controlLabel">
          Check all that apply to you: 
        </label>
        <div
          className="checkbox"
          style={undefined}>
          <label>
            <input
              disabled={false}
              name="hasWrittenPlan"
              onClick={[Function]}
              type="checkbox" />
            I have a written business plan
          </label>
        </div>
        <div
          className="checkbox"
          style={undefined}>
          <label>
            <input
              disabled={false}
              name="hasFinancialProjections"
              onClick={[Function]}
              type="checkbox" />
            I have financial projections
          </label>
        </div>
        <div
          className="checkbox"
          style={undefined}>
          <label>
            <input
              disabled={false}
              name="isGeneratingRevenue"
              onClick={[Function]}
              type="checkbox" />
            I\'m generating revenue
          </label>
        </div>
        <div
          className="checkbox"
          style={undefined}>
          <label>
            <input
              disabled={false}
              name="isVeteran"
              onClick={[Function]}
              type="checkbox" />
            I\'m a veteran
          </label>
        </div>
      </div>
      <button
        className="btn btn-default col-xs-2 col-xs-offset-5"
        type="submit">
        Continue 
      </button>
    </form>
  </div>
</div>
`;
